model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -358,14 +360,22 @@ def load_model_weights_sub_commands(
                 ]
             )
         else:
-            if framework == LLMInferenceFramework.TEXT_GENERATION_INFERENCE:
-                subcommands.append(
-                    f"{s5cmd} --numworkers 512 cp --concurrency 10 {os.path.join(checkpoint_path, '*')} {final_weights_folder}"
-                )
+            # Let's check whether to exclude "*.safetensors" or "*.bin" files
+            parsed_remote = parse_attachment_url(checkpoint_path)
+            all_files = s3_list_files(bucket=parsed_remote.bucket, key=parsed_remote.key)
+
+            # If there are more files ending in .safetensors, then exclude *.bin
```

Comment:
Also, is there a way to add a couple unit tests to make sure that this works? I think it would be great to add those if we can.

Reply:
Yep let me take a look at how that could make sense